File: 65.cpp
#include <regex>

struct Solution {
    const static regex wholePatt;

    bool isNumber(const string& s) {
        return regex_match(s, this->wholePatt);
    }
};
const regex Solution::wholePatt{regex(
    "^(?:(?:[+-]?\\d+)|(?:[+-]?(?:\\d+\\.\\d*|\\.\\d+)))(?:[eE][+-]?\\d+)?$"
)};

File: 65.py
import re

class Solution:
    intPatt = r'[+-]?\d+'
    decPatt = r'[+-]?(?:\d+\.\d*|\.\d+)'
    wholePatt = re.compile(f'^(?:(?:{intPatt})|(?:{decPatt}))(?:[eE]{intPatt})?$')

    def isNumber(self, s: str) -> bool:
        return self.wholePatt.match(s) is not None
